GalaxyZooTouchTable/GalaxyZooTouchTable/ViewModels/ClassificationPanelViewModel.cs

```diff
@@ -0,0 +1,174 @@
+        {
+            GetSubject();
+            LoadCommands();
+            Workflow = workflow;
+            CurrentTask = workflow.Tasks[workflow.FirstTask];
+            CurrentTaskIndex = workflow.FirstTask;
+
+            if (CurrentTask.Answers != null)
+            {
+                CurrentAnswers = ParseTaskAnswers(CurrentTask.Answers);
+            }
+        }
+
+        public event PropertyChangedEventHandler PropertyChanged;
+
+        private void OnPropertyRaised(string propertyname)
+        {
+            PropertyChanged(this, new PropertyChangedEventArgs(propertyname));
+        }
+
+        private void LoadCommands()
+        {
+            SelectAnswer = new CustomCommand(ChooseAnswer, CanSelectAnswer);
+            SubmitClassification = new CustomCommand(SendClassification, CanSendClassification);
+        }
+
+        private async void SendClassification(object obj)
+        {
+            CurrentClassification.Metadata.FinishedAt = DateTime.Now.ToString();
+            CurrentClassification.Annotations.Add(CurrentAnnotation);
+            ApiClient client = new ApiClient();
+            await client.Classifications.Create(CurrentClassification);
+            GetSubject();
+        }
+
+        private bool CanSendClassification(object obj)
+        {
+            return CurrentAnnotation != null;
+        }
+
+        private void ChooseAnswer(object obj)
+        {
+            AnswerButton button = (AnswerButton)obj;
+            SelectedItem = button;
+            CurrentAnnotation = new Annotation(CurrentTaskIndex, button.Index);
+        }
+
+        private bool CanSelectAnswer(object obj)
+        {
+            return CurrentAnswers.Count > 0;
+        }
+
+        public List<AnswerButton> ParseTaskAnswers(List<TaskAnswer> answers)
+        {
+            List<AnswerButton> renderedAnswers = new List<AnswerButton>();
+
+            for (int i = 0; i < answers.Count; i++)
+            {
+                AnswerButton item = new AnswerButton(answers[i], i);
+                renderedAnswers.Add(item);
+            }
+            return renderedAnswers;
+        }
+
+        public void StartNewClassification(Subject subject)
+        {
+            CurrentClassification = new Classification();
+            CurrentClassification.Metadata.WorkflowVersion = Workflow.Version;
+            CurrentClassification.Metadata.StartedAt = DateTime.Now.ToString();
+            CurrentClassification.Metadata.UserAgent = "Galaxy Zoo Touch Table";
+            CurrentClassification.Metadata.UserLanguage = "en";
+
+            CurrentClassification.Links = new ClassificationLinks(Config.ProjectId, Config.WorkflowId);
+            CurrentClassification.Links.Subjects.Add(subject.Id);
+
+            CurrentAnnotation = null;
+            SelectedItem = null;
+            CommandManager.InvalidateRequerySuggested();
+        }
+
+        private async void GetSubject()
+        {
+            if (Subjects.Count > 0)
```

Comment:
Not a huge deal, but you can change this to
```

if(Subjects.Count <= 0) 
{
	ApiClient client = new ApiClient();
    NameValueCollection query = new NameValueCollection
	{
		{ "workflow_id", Config.WorkflowId }
	};
	Subjects = await client.Subjects.GetList("queued", query);
}

SetSubject()
```

Also, you could then just get ride of `setSubject` and just put that code inline